compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/XCLBinGen.cpp

```diff
@@ -1183,15 +1344,28 @@ LogicalResult generateUnifiedObject(
       return failure();
     }
 
-    if (failed(runTool(
-            peanoLLCBin.string(),
-            {OptLLVMIRFile, "-O2", "--march=" + StringRef(targetArch).lower(),
-             "--function-sections", "--filetype=obj", "-o",
-             std::string(outputFile)},
-            verbose))) {
+    std::vector<std::string> llcArgs{OptLLVMIRFile,
+                                     "-O2",
+                                     "--march=" + StringRef(targetArch).lower(),
+                                     "--function-sections",
+                                     "--filetype=obj",
+                                     "-o",
+                                     outputFile,
+                                     "--stack-size-section"};
+
+    if (failed(runTool(peanoLLCBin.string(), llcArgs, verbose))) {
       llvm::errs() << "Failed to assemble ll with peano\n";
       return failure();
     }
+
+    // If this is not windows, we can do this check. On windows checkTool
+    // doesn't pipe outputs correctly.
```

Comment:
`outputs` being the stack sizes right?

Reply:
Yeah, in runTool the piping is gated on WIN32. Specifically I think it's this redirection for windows `redirects = {{}, {}, {}}` which means we don't get the logging back (and I don't want to sink hours into fixing this now!)

Reply:
(I've reworded)